packages/@react-spectrum/overlays/src/Modal.tsx

```diff
@@ -91,7 +91,7 @@ let ModalWrapper = forwardRef(function (props: ModalWrapperProps, ref: RefObject
   return (
     <div className={wrapperClassName}>
       <div
-        {...mergeProps(otherProps, overlayProps)}
+        {...mergeProps(otherProps, overlayProps, modalProps)}
```

Comment:
since ModalProps is only setting one value that the other two shouldn't have, I think this order of merging is okay.